dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,151 @@
+        if (string.Equals(modelId, "dall-e-2", StringComparison.OrdinalIgnoreCase))
         {
-            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+            this._logger.LogWarning($"{nameof(AzureOpenAITextToImageService)} supports only Dall-E-3. .");
+            throw new NotSupportedException("Dall-E-2 support was deprecated in Azure Open AI latest SDK. Please use a 'dall-e-3' deployment.");
         }
 
-        // Defaults if not supplied
-        maxRetryCount ??= 5;
-        apiVersion ??= "2023-06-01-preview";
-
-        this._core = new(httpClient, loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)));
-
-        this._endpoint = !string.IsNullOrEmpty(endpoint) ? endpoint! : httpClient!.BaseAddress!.AbsoluteUri;
-        this._apiKey = apiKey;
-        this._maxRetryCount = maxRetryCount.Value;
-        this._apiVersion = apiVersion;
-        this._core.AddAttribute(AIServiceExtensions.EndpointKey, endpoint);
-        this._core.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
-        this._core.AddAttribute(AIServiceExtensions.ApiVersionKey, apiVersion);
+        var connectorEndpoint = !string.IsNullOrWhiteSpace(endpoint) ? endpoint! : httpClient?.BaseAddress?.AbsoluteUri;
+        if (connectorEndpoint is null)
+        {
+            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+        }
 
-        this._core.RequestCreated += (_, request) => request.Headers.Add("api-key", this._apiKey);
+        this._client = new(new Uri(connectorEndpoint),
+            new AzureKeyCredential(apiKey),
+            GetClientOptions(httpClient, apiVersion));
     }
 
     /// <inheritdoc/>
-    public IReadOnlyDictionary<string, object?> Attributes => this._core.Attributes;
+    public IReadOnlyDictionary<string, object?> Attributes { get; } = new Dictionary<string, object?>();
 
     /// <inheritdoc/>
-    public async Task<string> GenerateImageAsync(string description, int width, int height, Kernel? kernel = null, CancellationToken cancellationToken = default)
+    public async Task<string> GenerateImageAsync(
+        string description,
+        int width,
+        int height,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
     {
-        var operationId = await this.StartTextToImageAsync(description, width, height, cancellationToken).ConfigureAwait(false);
-        var result = await this.GetTextToImageResultAsync(operationId, cancellationToken).ConfigureAwait(false);
+        Verify.NotNull(description);
+        const string SizeErr = "Dall-E 2 can generate only square images of size 256x256, 512x512, or 1024x1024";
```

Comment:
Since this class supports only Dall-E 3, this should be updated:
```suggestion
        const string SizeErr = "Dall-E 3 can generate only images of size 1024x1024, 1792x1024, or 1024x1792";
```